File: app/src/at-myhuynh/java/asiantech/internship/summer/java_core/ex08/ChainOfSymmetry.java
package asiantech.internship.summer.java_core.ex08;

import java.util.logging.Logger;

import asiantech.internship.summer.java_core.common.Common;

public class ChainOfSymmetry {

    public static void main(String[] args) {
        Logger log = Logger.getLogger(ChainOfSymmetry.class.getName());
        String str = Common.inputString("Nhập vào chuỗi: ");

        String[] arrStr = str.split("");
        int first = 0;
        int last = arrStr.length - 1;
        boolean check = true;

        while (first < last) {
            if (!arrStr[first++].equals(arrStr[last--])) {
                check = false;
                break;
            }
        }

        if (check) {
            log.info(str + ": Thuan nghich");
        } else {
            log.info(str + ": Khong Thuan nghich");
        }

        //Or Use StringBuffer
        log.info(String.valueOf(str.equals(new StringBuffer(str).reverse().toString())));

    }

}

File: app/src/at-myhuynh/java/asiantech/internship/summer/kotlin/ex08/ChainOfSymmetry.kt
package asiantech.internship.summer.kotlin.ex08

import asiantech.internship.summer.kotlin.common.Common
import asiantech.internship.summer.kotlin.common.isChainOfSymmetry

object ChainOfSymmetry {

    @JvmStatic
    fun main(args: Array<String>) {
        val str = Common.inputString("Nhập vào chuỗi: ")
        if (str.isChainOfSymmetry()) {
            println("$str Thuan nghich")
        } else {
            println("$str Khong thuan nghich")
        }
    }
}
